packages/opentelemetry-plugin-xml-http-request/src/xhr.ts

```diff
@@ -0,0 +1,371 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  B3Format,
+  BasePlugin,
+  isUrlIgnored,
+  isWrapped,
+  otperformance,
+} from '@opentelemetry/core';
+import * as tracing from '@opentelemetry/tracing';
+import * as types from '@opentelemetry/types';
+import {
+  addSpanNetworkEvent,
+  getResource,
+  parseUrl,
+  PerformanceTimingNames as PTN,
+  sameOriginOrUrlMatches,
+} from '@opentelemetry/web';
+import * as shimmer from 'shimmer';
+import { AttributeNames } from './enums/AttributeNames';
+import { EventNames } from './enums/EventNames';
+import {
+  OpenFunction,
+  PropagateTraceHeaderUrls,
+  SendBody,
+  SendFunction,
+  XMLHttpRequestWrapped,
+} from './types';
+
+/**
+ * XMLHttpRequest config
+ */
+export interface XMLHttpRequestPluginConfig extends types.PluginConfig {
+  // urls which should include trace headers when origin matches
+  propagateTraceHeaderUrls?: PropagateTraceHeaderUrls;
+}
+
+/**
+ * This class represents a XMLHttpRequest plugin for auto instrumentation
+ */
+export class XMLHttpRequestPlugin extends BasePlugin<XMLHttpRequest> {
```

Comment:
Could you add tests for this plugin? (Or maybe I missed it)

Reply:
there are tests for that

Reply:
Thanks, my bad on this - the file was collapsed by GitHub by default because it was big. Thanks for the detailed testing of all this!!